Classes/common/CDTDocumentRevision.m

```diff
@@ -79,11 +79,17 @@ + (CDTDocumentRevision *)createRevisionFromJson:(NSDictionary *)jsonDict
 
     // build the attachment objects
     for (NSString *key in [attachmentData allKeys]) {
-        NSURL *url = [NSURL URLWithString:key relativeToURL:documentURL];
+        
+        NSURLComponents *documentUrlComponents = [NSURLComponents componentsWithURL:documentURL resolvingAgainstBaseURL:NO];
+        NSString *documentPathString = documentUrlComponents.path;
+        NSString *attachmentPath = [NSString stringWithFormat:@"%@/%@", documentPathString,key];
+        documentUrlComponents.path = attachmentPath;
```

Comment:
the query is already in the `NSURLComponents` object, we are just replacing the path